### ISSUE TITLE
Failing AVIF and TIFF tests in `tests/limits.rs`

### ISSUE THREAD

**Issue by astral4** [image-rs/image#2100]
This happens with debug and release builds of version 0.24.8. I am using a M1 MacBook Pro with `rustc 1.77.0-nightly (2319be8e2 2024-01-12)`.

## Expected behavior

I expected all of the tests to pass.

## Actual behavior

Two tests fail: one for AVIF and one for TIFF. Both tests are located in `tests/limits.rs`.

Console output of testing the debug build (with `RUST_BACKTRACE=1`):

```
     Running tests/limits.rs (target/debug/deps/limits-02b8b99e31a91f0b)

running 7 tests
test bmp ... ok
test jpeg ... ok
test png ... ok
test webp ... ok
test gif ... ok
test tiff ... FAILED
test avif ... FAILED

failures:

---- tiff stdout ----
thread 'tiff' panicked at tests/limits.rs:179:76:
called `Result::unwrap()` on an `Err` value: Limits(LimitError { kind: InsufficientMemory })
stack backtrace:
   0: rust_begin_unwind
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/std/src/panicking.rs:647:5
   1: core::panicking::panic_fmt
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/result.rs:1649:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/result.rs:1073:23
   4: limits::tiff
             at ./tests/limits.rs:179:5
   5: limits::tiff::{{closure}}
             at ./tests/limits.rs:166:10
   6: core::ops::function::FnOnce::call_once
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/ops/function.rs:250:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

---- avif stdout ----
thread 'avif' panicked at src/codecs/avif/decoder.rs:59:9:
assertion `left == right` failed
  left: 10
 right: 8
stack backtrace:
   0: rust_begin_unwind
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/std/src/panicking.rs:647:5
   1: core::panicking::panic_fmt
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/panicking.rs:72:14
   2: core::panicking::assert_failed_inner
   3: core::panicking::assert_failed
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/panicking.rs:297:5
   4: image::codecs::avif::decoder::AvifDecoder<R>::new
             at ./src/codecs/avif/decoder.rs:59:9
   5: image::io::free_functions::load_decoder
             at ./src/io/free_functions.rs:55:59
   6: image::io::free_functions::load_inner
             at ./src/io/free_functions.rs:113:5
   7: image::io::free_functions::load
             at ./src/io/free_functions.rs:37:5
   8: image::dynimage::load_from_memory_with_format
             at ./src/dynimage.rs:1234:5
   9: limits::avif
             at ./tests/limits.rs:198:13
  10: limits::avif::{{closure}}
             at ./tests/limits.rs:193:10
  11: core::ops::function::FnOnce::call_once
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/ops/function.rs:250:5
  12: core::ops::function::FnOnce::call_once
             at /rustc/2319be8e265dd19973574eb581d28297baf44b11/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.


failures:
    avif
    tiff

test result: FAILED. 5 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 3.38s

error: test failed, to rerun pass `--test limits`
```

## Reproduction steps

1. Clone the repository
2. Run `cargo +nightly test --all-features` to test the debug build, or `cargo +nightly test --all-features --release` to test the release build.

**Comment by fintelia** [comment]
Thanks for the report!

The TIFF failure should be fixed by #2101. I'll look into the AVIF failure

**Comment by Shnatsel** [comment]
Ah, that's caused by my PR #2085. Sorry!
